File: sources/python/02_switch_case/main.py
# task 02 switch_case — expected output: 7500000075000000
# build: python main.py | pypy main.py | graalpy main.py | nuitka --standalone main.py    run: python main.py

acc = 0

for i in range(100000000):
    match i % 4:
        case 0:
            acc += 1
        case 1:
            acc += i
        case 2:
            acc += 2 * i
        case 3:
            acc += 3 * i

print(acc)

File: sources/rust/02_switch_case/main.rs
// task 02 switch_case — expected output: 7500000075000000
// build: rustc -O -o prog main.rs    run: ./prog

fn main() {
    let mut acc: u64 = 0;

    for i in 0..100_000_000u64 {
        match i % 4 {
            0 => acc += 1,
            1 => acc += i,
            2 => acc += 2 * i,
            _ => acc += 3 * i,
        }
    }

    println!("{}", acc);
}
